runtime/src/iree/hal/drivers/vulkan/util/intrusive_list.h

```diff
@@ -204,10 +204,26 @@ class IntrusiveListBase {
   size_t count_ = 0;
 };
 
+namespace impl {
+
+// std::iterator has been deprecated in C++17, therefore we create our own
+// definition of the required types for iterators.
+template <typename Category, typename T, typename Distance = std::ptrdiff_t,
+          typename Pointer = T*, typename Reference = T&>
+struct Iterator {
+  using iterator_category = Category;
+  using value_type = T;
+  using difference_type = Distance;
+  using pointer = Pointer;
+  using reference = Reference;
+};
+
+}  // namespace impl
+
 // Basic iterator for an IntrusiveList.
 template <typename T, size_t kOffset, bool kForward>
 class IntrusiveListIterator
-    : public std::iterator<std::input_iterator_tag, int> {
+    : public impl::Iterator<std::input_iterator_tag, int> {
```

Comment:
Since this has a single use in the whole codebase, should we just inline these typedefs inside `InstrusiveListIterator`?

Reply:
Very good point. Thank you for the feedback. 
I moved the definitions to InstrusiveListIterator, and modified the commit message.